### ISSUE TITLE
Keyboard dragging makes coil overlap on magnet just 1 step

### ISSUE THREAD

**Issue by zepumph** [phetsims/faradays-law#81]
It looks like on a keyboard you can do just 1 more step than with a mouse. @jessegreenberg do you have any thoughts about what is going on here? 

![image](https://user-images.githubusercontent.com/6856943/31287454-cfa5e526-aa6d-11e7-81c6-35a22790f33f.png)

**Comment by jessegreenberg** [comment]
I would guess that it is related to the fact that position is handled in ```moveMagnetToPosition```, which was designed to take the desired position from the mouse point, not the model position property. I am not sure yet if this is a shortcoming of the KeyboardDragHandler or the sim.

**Comment by jessegreenberg** [comment]
`KeyboardDragHandler` works like `MovableDragHandler`. The drag event directly updates a position Property in the handler.  Faraday's Law uses `SimpleDragHandler`, which doesn't set the position Property at all, but lets you pass in optional  listeners to be called at key points during the input. When using `KeyboardDragHandler` or `MovableDragHandler` with `FaradaysLawModel.moveMagnetToPosition`, the `position` param of  will always be equal to `this.magnetModel.positionProperty.value`, which is throwing off calculations in that function. `magnetBounds` wont be correct (that is probably what is causing this issue) and `movingDelta` will be `Vector2.ZERO`, which is probably why the magnet is sliding around the coils with the keyboard.

So brainstorming possible fixes:
- re-implement `moveMagnetToPosition` so that it doesn't need a desiredPosition from the pointer?
- Create a new drag handler type like `SimpleKeyboardDragHandler` that works better for things that are using `SimpleDragHandler`?

I am  liking the second option, and KeyboardDraghandler could extend SimpleKeyboardDragHandler (or maybe it should have a different name). @zepumph do you agree with this, or can you think of alternatives?

**Comment by zepumph** [comment]
I'm unsure about the best option here.  I can imagine that we will run into this a lot, and it would be nice if normal and keyboard drag handlers hook up to sims identically (or as identically as possible). I'd say go for it. I would love to assist/review/pair with you, but won't be able to until at least next week.

**Comment by jessegreenberg** [comment]
I verified that the following hack fixes things:

Change `onDrag` in `MagnetNodeWithField`'s `KeyboardDragHandler` to this
```js
        onDrag: function( event ) {

          var desiredPosition = event.point;
          model.moveMagnetToPosition( desiredPosition );
        }
```

And replace lines 213-216 of KeyboardDragHandler with
```js
          this.onDrag( { point: newPosition } );
```
(removing `this.positionProperty.set( newPosition );`)

**Comment by zepumph** [comment]
Looks like you got it. Let me know if I can help.

**Comment by jessegreenberg** [comment]
> I can imagine that we will run into this a lot

I agree, we could be re-implementing most listeners that use SimpleDragHandler, that would be bad! Thanks @zepumph, moving this issue to scenery,  but I likely won't  get to this for a while either.

**Comment by jessegreenberg** [comment]
This might have been fixed with #716, I should look into this and see if this is still a bug.

**Comment by jessegreenberg** [comment]
This is no longer high priority. @mbarlow12 is this still an issue in Faradays Law or in the a11y branch you are working in?

**Comment by mbarlow12** [comment]
I can't get this issue to appear in master after recent commits/merges. @zepumph feel free to review and close.
